editor/src/clj/editor/debugging/mobdebug.clj

```diff
@@ -149,17 +149,20 @@
     (catch java.lang.Exception _ nil)))
 
 (defn connect!
-  [address port on-connected on-closed]
+  [address port on-connected on-closed on-error]
   (thread
-    (loop [retries 0]
-      (if-some [socket (try-connect! address port)]
-        (let [debug-session (make-debug-session socket on-closed)]
-          (on-connected debug-session))
-        (if (< retries 50)
-          (do (Thread/sleep 200) (recur (inc retries)))
-          (throw (ex-info (format "Failed to connect to debugger on %s:%d" address port)
-                          {:address address
-                           :port port})))))))
+    (try
+      (loop [retries 0]
+        (if-some [socket (try-connect! address port)]
+          (let [debug-session (make-debug-session socket on-closed)]
+            (on-connected debug-session))
+          (if (< retries 50)
+            (do (Thread/sleep 200) (recur (inc retries)))
+            (throw (ex-info (format "Failed to connect to debugger on %s:%d" address port)
+                            {:address address
+                             :port port})))))
+      (catch Exception e
+        (on-error e)))))
```

Comment:
this is the main fix. 
because of the thread, we don't catch the exception in `start-debugger!`, so I use `on-error` callback instead